### PR TITLE
update ingress version in ks-core chart

### LINKED ISSUE #4283: Failed to deploy sample App
**Describe the Bug**
When Gateway is enabled, Sample App can't be deployed. The Ingress version doesn't match with server side.

For UI issues please also add a screenshot that shows the issue.
ingress

**Versions Used**
KubeSphere:
Kubernetes: (If KubeSphere installer used, you can skip this)

**Environment**
How many nodes and their hardware configuration:

For example: CentOS 7.5 / 3 masters:  8cpu/8g; 3 nodes: 8cpu/16g
(and other info are welcomed to help us debugging)

**How To Reproduce**

**Expected behavior**

### PR DESCRIPTION
**Pull request by RolandMa1986**
Signed-off-by: Roland.Ma <rolandma@kubesphere.io>

### What type of PR is this?
/kind bug
/kind flake
### What this PR does / why we need it:
Since #4068 we have upgraded the Ingress API versions to networking.k8s.io/v1. but the sample app still uses the old API version. So we have to upgrade it to the networking.k8s.io/v1 version.

### Which issue(s) this PR fixes:
Fixes #4283

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?
```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by yuswift** [comment]
ks-installer should be updated too.

File: config/ks-core/templates/ks-apiserver.yml [hunk 1]
```diff
@@ -43,12 +43,14 @@ spec:
         volumeMounts:
         - mountPath: /var/run/docker.sock
```
Comment (zryfish) [inline config/ks-core/templates/ks-apiserver.yml:44]: I think we can remove `docker.sock` in v3.2, since there are no components depending on this.

**Pull request merged**